frontend/src/Admin/components/ShowJudgesTab.js

```diff
@@ -1,5 +1,90 @@
 import React from 'react'
+import PropTypes from 'prop-types'
+import { Row, Col, Progress, Button } from 'reactstrap'
+import { Link } from 'react-router-dom'
 
-export default function () {
-  return 'Judges'
+// Calculates a Bootstrap color for the given fractional progress, given
+// as a number from 0 to 1.
+const colorForProgress = progress => {
+  if (progress < 0.1) {
+    return 'danger'
+  }
+  if (progress < 0.25) {
+    return 'warning'
+  }
+  if (progress < 1) {
+    return null // default color
+  }
+  return 'success'
 }
+
+const ShowJudgesTab = ({ show }) => {
+  // Extract all votes we care about
+  const allVotes = show.entries.reduce(
+    // Don't count votes on entries that are excluded
+    (lst, entry) => (entry.excludeFromJudging ? lst : [...lst, ...entry.votes]),
+    []
+  )
+  // Count the votes by judge
+  const voteCountByJudge = allVotes.reduce(
+    (accum, vote) => ({
+      ...accum,
+      [vote.judge.username]: (accum[vote.judge.username] || 0) + 1
+    }),
+    {}
+  )
+  // Count the number of total entries that are in judging
+  const numEntries = show.entries.filter(entry => !entry.excludeFromJudging)
+    .length
+
+  return (
+    <div>
+      <Row style={{ marginBottom: '1.5rem' }}>
+        <Col className='text-right'>
+          <Link to={`/show/${show.id}/judges/assign`}>
+            <Button color='primary'>Manage Judges</Button>
+          </Link>
+        </Col>
+      </Row>
+      {show.judges.map(judge => {
+        const numVotes = voteCountByJudge[judge.username] || 0
+        return (
+          <Row key={judge.username} style={{ marginBottom: '1.5rem' }}>
+            <Col md='3'>
+              {judge.firstName} {judge.lastName} ({judge.username})
+            </Col>
+            <Col>
+              <Progress
+                value={numVotes / numEntries * 100}
+                color={colorForProgress(numVotes / numEntries)}
+              >
+                {numVotes} / {numEntries}
```

Comment:
On Judges who haven't started voting yet, this currently just shows a "/"

![](https://user-images.githubusercontent.com/13719429/38175416-31e448d2-35aa-11e8-9dd3-64f9914f01c7.png)

Should show 0 / <total>, ideally in the center of the bar in black.

Reply:
Hmm... I think in this case it's easier to just not show it at all. The CSS for the former would be hacky I suspect... since I don't think there's a markup for it